cmd/manager/deployments.go

```diff
@@ -479,23 +489,23 @@ func getRegistry(w http.ResponseWriter, r *http.Request, handler string) *common
 	return t
 }
 
-func createRegistryHandlerFunc(w http.ResponseWriter, r *http.Request, c *router.Context) error {
-	handler := "manager: create registry"
+func createRepoHandlerFunc(w http.ResponseWriter, r *http.Request, c *router.Context) error {
+	handler := "manager: create repository"
 	util.LogHandlerEntry(handler, r)
 	defer r.Body.Close()
-	registryName, err := pos(w, r, 2)
+	repoName, err := pos(w, r, 2)
 	if err != nil {
 		return err
 	}
 
-	reg := getRegistry(w, r, handler)
-	if reg.Name != registryName {
-		e := fmt.Errorf("Registry name does not match %s != %s", reg.Name, registryName)
+	reg := getRepo(w, r, handler)
+	if reg.Name != repoName {
+		e := fmt.Errorf("Repo name does not match %s != %s", reg.Name, repoName)
 		httputil.BadRequest(w, r, e)
 		return nil
 	}
 	if reg != nil {
-		err = c.Manager.CreateRegistry(reg)
+		err = c.Manager.CreateRepo(reg)
```

Comment:
It takes the chart as a payload, not just the URL. Per our discussion today, I will unify the two as AddChartRepo.